programs/compressed-token/src/burn.rs

```diff
@@ -27,13 +34,86 @@ pub struct CompressedTokenInstructionDataBurn {
     pub delegated_transfer: Option<DelegatedTransfer>,
 }
 
+#[derive(Accounts)]
+pub struct BurnInstruction<'info> {
+    #[account(mut)]
+    pub fee_payer: Signer<'info>,
+    pub authority: Signer<'info>,
+    /// CHECK: that mint authority is derived from signer
+    #[account(seeds = [CPI_AUTHORITY_PDA_SEED], bump,)]
+    pub cpi_authority_pda: UncheckedAccount<'info>,
+    /// CHECK: that authority is mint authority
+    #[account(mut, constraint = mint.mint_authority.unwrap() == authority.key())]
+    pub mint: Account<'info, Mint>,
```

Comment:
we probably don't need to pass mint as account here just in the instruction data.
You can probably just copy and extend `pub struct GenericInstruction<'info> ` with the  `token_pool_pda`